lib/active_model/serializer/association.rb

```diff
@@ -10,24 +10,111 @@ class Serializer
     #  Association.new(:comments, { serializer: CommentSummarySerializer })
     #
     class Association < Field
+      # association_options =  {
+      #   key:,
+      #   parent_serializer: ,
+      #   parent_serializer_options:
+      #   include_slice:
+      # }
+      # Build association. This method is used internally to
+      # build serializer's association by its reflection.
+      #
+      # @param [Serializer] parent_serializer for given association
+      # @param [Hash{Symbol => Object}] parent_serializer_options
+      #
+      # @example
+      #    # Given the following serializer defined:
+      #    class PostSerializer < ActiveModel::Serializer
+      #      has_many :comments, serializer: CommentSummarySerializer
+      #    end
+      #
+      #    # Then you instantiate your serializer
+      #    post_serializer = PostSerializer.new(post, foo: 'bar') #
+      #    # to build association for comments you need to get reflection
+      #    comments_reflection = PostSerializer._reflections.detect { |r| r.name == :comments }
+      #    # and #build_association
+      #    comments_reflection.build_association(post_serializer, foo: 'bar')
+      #
+      # @api private
+      def self.from_reflection(reflection, association_options)
+        block = nil
+        Association.new(reflection.name, association_options.merge(reflection: reflection), block)
+      end
+
+      attr_reader :parent_serializer, :parent_serializer_options, :include_slice, :reflection, :association_options
+      def initialize(*)
+        super
+        @association_options = options
+        @parent_serializer = association_options.fetch(:parent_serializer)
+        @parent_serializer_options = association_options.fetch(:parent_serializer_options)
+        @include_slice = association_options.fetch(:include_slice)
+        @reflection = association_options.fetch(:reflection)
```

Comment:
and [https://github.com/rails/rails/blob/master/activerecord/lib/active_record/reflection.rb](https://github.com/rails/rails/blob/050018d48e2c94a36cb0ec61288b1c5519059d36/activerecord/lib/active_record/reflection.rb) has some interesting ideas in it